### ISSUE TITLE
Remove replace directives from go.mod file

### ISSUE THREAD

**Issue by deelawn** [prysmaticlabs/prysm#10046]
### Background
Other go projects that import prysm modules may run into issues due to `replace`  in the go.mod file.  It seems like it is generally best to avoid using `replace` in the go.mod file of published modules.  Best case is that the consumer will need to include the replacement in their own go.mod file in order for the build to succeed.  Worst case is that the build succeeds but doesn't include potentially critical changes that have been applied using `replace`.

### Description
For example, take this line from the current go.mod file:
```replace github.com/ferranbt/fastssz => github.com/prysmaticlabs/fastssz v0.0.0-20211123050228-97d96f38caae```

Any consumers that import a prysm module with this dependency will not be able to successfully build their program until they copy and paste this line into their own go.mod file.  This is because `replace` directives are only applied to the top level module, in this case being the consumer's own module.

Another case to consider:
Say for example that the fastssz replacement referenced above _didn't_ change the API and the build succeeded.  Now, any consumers of a prysm module that includes fastssz as a dependency will be running code from the original repository while tests written and committed to the prysm repository will be using the replaced module.

So, can these be removed and the import paths updated throughout the repository?  Is there a reason to keep the replace directives other than the convenience of not updating import paths?

**Comment by prestonvanloon** [comment]
I agree with this sentiment. We should remove replace directives and either merge changes upstream or actually replace the imports with a forked repository.
